meal-planning/src/components/recipe-library/RecipesLibrary.js

```diff
@@ -26,20 +26,24 @@ const RecipesLibrary = () => {
     return recipe.name.toLowerCase().includes(search.toLowerCase());
   });
 
-  const openModal = () => {
-    setIsModalOpen(true);
+  const handleAddRecipeDialogClose = () => {
+    setAddRecipeDialogOpen(false);
   };
 
-  const handleDialogClose = () => {
-    setIsModalOpen(false);
+  const handleAddRecipeDialogSubmit = recipe => {
+    setRecipes(recipes.concat(recipe));
   };
 
-  const handleDialogSumbit = recipe => {
-    setIsModalOpen(false);
-    setRecipes(recipes.concat(recipe));
+  const handleEditRecipeDialogSubmit = editedRecipe => {
+    const originalRecipeIndex = recipes.findIndex(
```

Comment:
could probably use a null check/error handling here in case something weird happened